amplify/function/BusinessLogic/GenerateTeamCode/handler.ts

```diff
@@ -0,0 +1,135 @@
+        region: process.env.AWS_REGION,
+        defaultAuthMode: "iam",
+      },
+    },
+  },
+  {
+    Auth: {
+      credentialsProvider: {
+        getCredentialsAndIdentityId: async () => ({
+          credentials: {
+            accessKeyId: process.env.AWS_ACCESS_KEY_ID as string,
+            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY as string,
+            sessionToken: process.env.AWS_SESSION_TOKEN as string,
+          },
+        }),
+        clearCredentialsAndIdentityId: () => {
+          /* noop */
+        },
+      },
+    },
+  },
+);
+
+const client = generateClient<Schema>({
+  authMode: "iam",
+});
+
+export const handler: Schema["GenerateTeamCode"]["functionHandler"] = async (
+  event,
+) => {
+  let team = null;
+  let teamId: string | null = null;
+  try {
+    do {
+      teamId = Array.from(Array(4), () =>
+        Math.floor(Math.random() * 36).toString(36),
+      ).join("");
+
+      team = (
+        await client.graphql({
+          query: getTeam,
+          variables: {
+            id: teamId,
+          },
+        })
+      ).data.getTeam;
+    } while (team != null);
+
+    const teamCreation = await client
+      .graphql({
+        query: createTeam,
+        variables: {
+          input: {
+            name: event.arguments.teamName,
+            id: teamId,
+          },
+        },
+      })
+      .catch(() => {
+        return {
+          body: { value: `Error creating team` },
+          statusCode: 500,
+          headers: { "Content-Type": "application/json" },
+        };
+      });
+
+    if (teamCreation) {
+      if (event.arguments.addCallerToTeam) {
+        await client
+          .graphql({
+            query: updateUser,
+            variables: {
+              input: {
+                id: (event.identity as AppSyncIdentityCognito).sub,
+                teamId: teamId,
+              },
+            },
+          })
+          .then(() => {
+            return {
+              body: { value: teamId },
+              statusCode: 200,
+              headers: { "Content-Type": "application/json" },
+            };
+          })
+          .catch(() => {
+            return {
+              body: { value: `Error updating user (team was created)` },
+              statusCode: 500,
+              headers: { "Content-Type": "application/json" },
+            };
+          });
+      } else {
+        return {
+          body: { value: teamId },
+          statusCode: 200,
+          headers: { "Content-Type": "application/json" },
+        };
+      }
+    } else {
+      return {
+        body: { value: `Error creating team` },
+        statusCode: 500,
+        headers: { "Content-Type": "application/json" },
+      };
+    }
+  } catch {
+    return {
+      body: { value: `Unhandled Internal Server Error` },
+      statusCode: 500,
+      headers: { "Content-Type": "application/json" },
+    };
+  }
+  return {
+    body: { value: `No return condition reached` },
```

Comment:
I didn't wanna do this but it seems like the linter doesn't recognize that the .then() or the .catch() would be hit and then complains that the function might not return anything